docs/dev-notes/DEVELOPMENT.md

```diff
@@ -0,0 +1,135 @@
+
+Use the following commands to run the app
+
+```
+npx react-native run-android
+```
+
+Android app will be launched on physical device if connected using `adb`, if not, it should open an emulator and run the app on it. However, if you have multiple devices connected, you can run on a specific device by running following command:
+
+```
+npx react-native run-android --deviceId=DEVICE_ID
+```
+
+Where `DEVICE_ID` can be obtained by running `adb devices` command.
+
+#### Option 2: Using Android Studio
+
+- Open the `android` folder in Android Studio
+- Build the app in Android Studio
+- Run the app
+
+### iOS
+
+*Note: It's assumed that you have XCode installed on your machine before continuing*
+
+iOS app requires installing libraries using CocoaPods. You can install it by running following commands
+
+```
+cd ios
+pod install
+cd ..
+```
+
+Next, follow these steps:
+
+- Jump into `ios` folder and open [`.xcworkspace`](/ios/SampleApp.xcworkspace) file to setup the project
+- [Configure iOS Code Signing](https://github.com/customerio/mobile/blob/main/ios_code_signing.md#how-do-i-setup-my-computer-for-code-signing) to download the provisioning profiles and certificates for development to your machine. 
+
+> **Note:**
+> Make sure that the deployment target is set to the minimum [required by the iOS SDK](https://github.com/customerio/customerio-ios#readme).
+
+#### Option 1: Using Command Line
+
+**To run the app on iOS Simulator**
+
+Use the following commands to run the app on iOS Simulator
+
+```
+npx react-native run-ios
+```
+**To run the app on device**
+
+To run using terminal
+
+```
+npx react-native run-ios --device "Your iPhone"
+```
+
+#### Option 2: Using XCode
+
+- Open [`.xcworkspace`](ios/SampleApp.xcworkspace) file in `ios` folder
+- Select the device from the list of devices available
+- Run the app 
+
+# Work on SDK locally
+
+Did you just implement a bug fix or new feature to the [React Native SDK](https://github.com/customerio/customerio-reactnative) that you want to test your change locally in Ami App? Want to work on Ami App *and* the SDK at the same time on your local machine? You can! 
+
+1. Follow the [SDK docs](https://github.com/customerio/customerio-reactnative/blob/docs/dev-env/docs/dev-notes/DEVELOPMENT.md#work-on-amiapp-locally) to setup your SDK for Ami App to install.
+2. Changes made in JS/TS file will be visible instantly. However, if you made any changes to native code, you will need to re-run the app. We recommend removing `node_modules` folder using `rm -rf node_modules` before running the app again.
+3. Run `yarn install` to update code from dependencies if needed.
+4. Run the Ami App following the instructions above.
+
+> **Tip**
+> 
+> _You can combine commands when running the app after making changes to native code
+> <br/>e.g.
+> <br/>*Android:* `rm -rf node_modules && yarn install && npx react-native run-android`
+> <br/>*iOS:* `rm -rf node_modules && yarn install && npx react-native run-ios`_
```

Comment:
Some projects on our mobile team [leverage a `Makefile` to make handy scripts](https://github.com/customerio/customerio-ios/blob/develop/Makefile) (you can also use `package.json` file I suppose). 

We could make a script `update-native-ios` that does all of these steps for you.

Reply:
Sounds a good idea to me, I'll see how can we utilize it make the development easier for this package and update docs accordingly.

Reply:
Added scripts and instructions for it